src/Renderer/LayeredMaterial.js

```diff
@@ -41,288 +43,400 @@ var getColorAtIdUv = function(nbTex) {
+            if (this.textures[l][i] instanceof THREE.Texture) {
+                this.textures[l][i].dispose();
+            }
+        }
+    }
 
     var jT = new JavaTools();
 
-    jT.freeArray(this.Textures[0]);
-    jT.freeArray(this.Textures[1]);
+    jT.freeArray(this.textures[0]);
+    jT.freeArray(this.textures[1]);
 
     jT.freeArray(this.uniforms.dTextures_00.value);
     jT.freeArray(this.uniforms.dTextures_01.value);
 };
 
-LayeredMaterial.prototype.nbLoadedTextures = function() {
+LayeredMaterial.prototype.setSequence = function(sequenceLayer) {
 
-    return this.nbTextures[0] + this.nbTextures[1];
-};
+    let offsetLayer = 0;
+    let offsetTexture = 0;
 
+    for (let l = 0; l < sequenceLayer.length; l++) {
 
-LayeredMaterial.prototype.setSequence = function(newSequence) {
+        let layer = sequenceLayer[l];
+        let oldIndex = this.indexOfColorLayer(layer);
+        if(oldIndex>-1) {
 
-    var sequence = this.uniforms.layerSequence.value;
-    var max = Math.min(newSequence.length, sequence.length);
+            let newIndex = l-offsetLayer;
+            let texturesCount = this.layerTexturesCount[oldIndex];
 
-    for (var l = 0; l < max; l++)
+            if(newIndex !== oldIndex) {
 
-        sequence[l] = newSequence[l];
+                moveElementArray(this.colorLayersId,oldIndex,newIndex);
+                moveElementArray(this.layerTexturesCount,oldIndex,newIndex);
+                moveElementArray(this.uniforms.paramLayers.value,oldIndex,newIndex);
+                moveElementArray(this.uniforms.visibility.value,oldIndex,newIndex);
+                let oldOffset = this.getTextureOffsetByLayerIndex(newIndex);
+                moveElementsArray(this.uniforms.offsetScale_L01.value,oldOffset,texturesCount,offsetTexture);
+                moveElementsArray(this.uniforms.dTextures_01.value,oldOffset,texturesCount,offsetTexture);
+            }
 
-};
+            this.setTextureOffsetByLayerIndex(newIndex,offsetTexture);
+            offsetTexture += texturesCount;
+
+        } else {
+            offsetLayer++;
+        }
+    }
 
-LayeredMaterial.prototype.removeLayerColor = function(idLayer) {
-    var layerIndex = this.layerIdToIndex[idLayer];
-    if (!layerIndex) return;
+    this.uniforms.colorLayersCount.value = this.getColorLayersCount();
+};
 
-    var startIdTexture = this.paramLayers[layerIndex].x;
-    var nbTextures = this.getNbColorTexturesLayer(layerIndex);
+LayeredMaterial.prototype.removeColorLayer = function(layer) {
 
-    this.paramLayers.splice(layerIndex, 1);
-    this.paramBLayers.splice(layerIndex, 1);
-    this.paramLayers.push(vector4);
-    this.paramBLayers.push(vector2);
+    let layerIndex = this.indexOfColorLayer(layer);
 
-    for (var i = startIdTexture, max = startIdTexture + nbTextures; i < max; i++) {
-        if (this.Textures[1][i] instanceof THREE.Texture)
-            this.Textures[1][i].dispose();
+    if (layerIndex === -1) {
+        return;
```

Comment:
At some point we'll need to stop silently ignoring errors and decide on a strategy (log errors ? throw ?)